### PR TITLE
PWA-2543: Parent product image thumbnail has been showns in mini cart instead of product itself

### PR DESCRIPTION
**Pull request by rwalter215**
## Description

- Refactored configuredVariant util to reflect updated graphql structure
- Created unit tests for configuredVariant
- Added JSDoc information for configuredVariant

## Related Issue

Closes PWA-2543 

### Verification Stakeholders

PM/UX (Approval granted)

## Verification Steps

Test plan outlined in: 
https://jira.corp.magento.com/browse/PWA-2543

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by mikhaelbois — DISMISSED** [review]
LGTM ✅

**Comment by eug123** [comment]
@rwalter215,
Initial issue with configured variant thumbnail is fixed for minicart. 

Per description new cypress mock should be added in the scope of this ticket, is this scenario is covered with cypress tests?

**Comment by rwalter215** [comment]
Currently working on the Cypress tests. I also still need to get UX/PM approval

**Comment by rwalter215** [comment]
I updated cypress integration tests, ready for re-review

File: packages/venia-ui/lib/components/MiniCart/ProductList/item.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ const Item = props => {
                             ? configured_variant.thumbnail.url
                             : product.thumbnail.url
                     }
+                    data-cy="item-image"
```
Comment (pedrochiossi) [inline packages/venia-ui/lib/components/MiniCart/ProductList/item.js:75]: We always start with upper case for `data-cy` 
```suggestion
                    data-cy="Item-image"
```

**Comment by rwalter215** [comment]
Made requested changes

**Review by pedrochiossi — APPROVED** [review]

**Comment by mikhaelbois** [comment]
@rwalter215 
1. Getting the following error in the cart.

image

To resolve, you can apply the following:
```
// packages/venia-ui/lib/components/CartPage/ProductListing/productListing.js

const productComponents = items.map((product, key) => (
    <Product
        key={key}
        item={product}
        setActiveEditItem={setActiveEditItem}
        setIsCartUpdating={setIsCartUpdating}
        onAddToWishlistSuccess={onAddToWishlistSuccess}
        fetchCartDetails={fetchCartDetails}
        wishlistConfig={wishlistConfig}
    />
));
```
2. Wishlist items image does not reflect the configurations. If out of scope, could you create a follow-up task?
 
image

3. The failing Cypress tests already have a follow-up task. ✅

**Pull request merged**